pyarr/base.py

```diff
@@ -47,13 +60,65 @@ def assert_return(
         assert isinstance(response, typearg)
         return response
 
+    def assert_return_put(
+        self,
+        path: str,
+        ver_uri: str,
+        typearg: type,
+        params: Optional[dict] = None,
+        data: Optional[Union[dict[str, Any], list[dict[str, Any]]]] = None,
+    ) -> Any:
+        """Helper function to add assert to enforce typing responses
+
+        Args:
+            path (str): Path on API
+            ver_uri (str): API Version
+            typearg (type): Python Type
+            params (Union[dict[str, Any], None], optional): Any required params. Defaults to None.
+
+        Returns:
+            Any: Any
+        """
+        if params is None:
+            response = self._put(path, ver_uri, data=data)
+        else:
+            response = self._put(path, ver_uri, params=params, data=data)
+        assert isinstance(response, typearg)
+        return response
+
+    def assert_return_post(
+        self,
+        path: str,
+        ver_uri: str,
+        typearg: type,
+        params: Optional[dict] = None,
+        data: Optional[Union[dict[str, Any], list[dict[str, Any]]]] = None,
+    ) -> Any:
+        """Helper function to add assert to enforce typing responses
+
+        Args:
+            path (str): Path on API
+            ver_uri (str): API Version
+            typearg (type): Python Type
+            params (Union[dict[str, Any], None], optional): Any required params. Defaults to None.
+
+        Returns:
+            Any: Any
+        """
+        if params is None:
+            response = self._post(path, ver_uri, data=data)
+        else:
+            response = self._post(path, ver_uri, params=params, data=data)
+        assert isinstance(response, typearg)
```

Comment:
This was done mainly because it was the only way I could figure to make sure the correct types were returned, without this typing errors were being issued. There may be a better way to handle this for sure.